core/src/main/scala/codesearch/core/index/LanguageIndex.scala

```diff
@@ -6,72 +6,63 @@ import ammonite.ops.{Path, root}
 
 import sys.process._
 import codesearch.core.db.DefaultDB
-import codesearch.core.model.DefaultTable
+import codesearch.core.model.{DefaultTable, Version}
 import codesearch.core.util.Helper
 import org.apache.commons.io.FilenameUtils
 import org.slf4j.Logger
 
 import scala.collection.mutable
 import scala.concurrent.{ExecutionContext, Future}
 
-trait Sources[VTable <: DefaultTable] {
+trait LanguageIndex[VTable <: DefaultTable] {
   protected val indexAPI: Index with DefaultDB[VTable]
   protected val logger: Logger
   protected val langExts: String
 
   protected val indexFile: String
   protected lazy val indexPath: Path = root / 'root / 'aelve / 'data / indexFile // FIXME
 
-  def downloadSources(name: String, ver: String): Future[Int]
-
-  implicit val ec: ExecutionContext = new ExecutionContext {
-
-    import java.util.concurrent.Executors
-
-    private val threadPool = Executors.newFixedThreadPool(100)
-
-    override def execute(runnable: Runnable): Unit = {
-      threadPool.submit(runnable)
-    }
-
-    override def reportFailure(cause: Throwable): Unit = {
-      cause.printStackTrace()
-    }
-  }
+  def downloadMetaInformation(): Unit = indexAPI.updateIndex()
 
+  def getLastVersions: Map[String, Version] = indexAPI.getLastVersions
 
-  def update(): Future[Int] = {
-    val lastVersions = indexAPI.getLastVersions.mapValues(_.verString)
-
-    val futureAction = indexAPI.verNames().flatMap { packages =>
-      val packagesMap = Map(packages: _*)
-
-      Future.sequence(lastVersions.filter {
-        case (packageName, currentVersion) =>
-          !packagesMap.get(packageName).contains(currentVersion)
-      }.map {
-        case (packageName, currentVersion) =>
-          downloadSources(packageName, currentVersion)
-      })
-    }.map(_.sum)
+  def downloadSources(name: String, ver: String): Future[Int]
 
-    Future {
-      logger.debug("UPDATE PACKAGES")
-    } flatMap { _ =>
-      futureAction
-    }
+  implicit def executor: ExecutionContext
+
+  def updatePackages(): Future[Int] = {
```

Comment:
Same – some nameless `Int` and a method without a docstring. (Ideally, all new methods should have a docstring)